### PR TITLE
#159053983 Reset password via email

### PR DESCRIPTION
**Pull request by zakhusayn**
**What does this PR do?**
Reset password using via email.

**Description of Task to be completed?**

* Sends a verification email to the user to reset password.
* Update user's password

**Endpoint**

| Method       | Endpoint          | Functionality  |
| ------------- |:-------------:| -----:|
| POST      | api/reset-password/|          Sends email to the user |
|POST      | api/rest-password-done/ | Update password |

**How should this be manually tested?**
* clone this repository
* Update .env file with sendgrid email configuration
`cd ah-scorpion
pip install -r requirements.txt
python manage.py makemigrations
python manage.py migrate
`
* Launch Postman App.
* Create an account in postman

```
URL: http://127.0.0.1:8000/api/users/

{
  "user": {
	"username":"john_doe",
	"email":"john_doe@example.com",
	"password":"Password1234"
  }
}
```
* Post the email your registered with to be sent an email and follow instruction.

```
URL: http://127.0.0.1:8000/api/reset-password/

{"email": john_doe@example.com}

```

* Update your password

```
URL: http://127.0.0.1:8000/api/reset-password-done/

{"new_password": "1234567789",
 "email":"john_doe@gmail.com",
 "reset_token": "4yf-62dbe3577b5f18a7eb53"}

```

**What are the relevant pivotal tracker stories?**

[#159053983](https://www.pivotaltracker.com/story/show/159053983)

### CODE REVIEW

File: authors/apps/authentication/migrations/0001_initial.py [hunk 1]
```diff
@@ -1,4 +1,8 @@
+<<<<<<< HEAD
```
Comment (RuthNjeri) [inline authors/apps/authentication/migrations/0001_initial.py]: Remove this line

File: authors/apps/authentication/migrations/0001_initial.py [hunk 2]
```diff
@@ -1,4 +1,8 @@
...
+# Generated by Django 2.0.6 on 2018-08-03 09:04
+=======
```
Comment (RuthNjeri) [inline authors/apps/authentication/migrations/0001_initial.py]: Remove this as well

File: authors/apps/authentication/migrations/0001_initial.py [hunk 3]
```diff
@@ -1,4 +1,8 @@
...
 # Generated by Django 2.0.6 on 2018-07-31 11:59
+>>>>>>> c1f6dd1d384f8339aed9a5a5024b0ba424b02d07
```
Comment (RuthNjeri) [inline authors/apps/authentication/migrations/0001_initial.py]: Remove this also

**Review by RuthNjeri — COMMENTED** [review]
During your tests, when creating a user, you can update the field isActive so that your login tests do not fail then you can proceed to password reset.

**Comment by RuthNjeri** [comment]
when I run the tests I am getting this error `Reverse for 'reset-password' not found. 'reset-password' is not a valid view function or pattern name`

You can put the names of the urls in the urls.py file

**Comment by dmigwi** [comment]
Great work Zakaria. This PR worked flawlessly.

**Pull request merged**